syntax = "proto3";

package v3;

option go_package = "github.com/dendrascience/dendra-data-api/release/go/v3";

import "v3/types.proto";

service ConverterService {
  // NOTES:
  // rpc ListUnits

  rpc ConvertAggregates (ConvertAggregatesRequest) returns (ConvertAggregatesResponse) {}
  rpc ConvertDatapoints (ConvertDatapointsRequest) returns (ConvertDatapointsResponse) {}
}

message ConvertAggregatesRequest {
  ConvertArgs convert = 1;
  repeated Aggregate aggregates = 2;
}

message ConvertAggregatesResponse {
  repeated Aggregate aggregates = 1;
}

message ConvertDatapointsRequest {
  ConvertArgs convert = 1;
  repeated Datapoint datapoints = 2;
}

message ConvertDatapointsResponse {
  repeated Datapoint datapoints = 1;
}
